store/src/main/java/org/apache/rocketmq/store/MappedFileQueue.java

```diff
@@ -204,14 +232,23 @@ public MappedFile getLastMappedFile(final long startOffset, boolean needCreate)
         }
 
         if (createOffset != -1 && needCreate) {
-            String nextFilePath = this.storePath + File.separator + UtilAll.offset2FileName(createOffset);
-            String nextNextFilePath = this.storePath + File.separator
-                + UtilAll.offset2FileName(createOffset + this.mappedFileSize);
+            String nextFilePath;
+            String nextNextFilePath;
+            if (config != null && config.isMultiCommitLogPathEnable()) {
+                long fileIdx = createOffset / this.mappedFileSize;
+                List<String> pathList = config.getStorePathCommitLogList();
+                nextFilePath = pathList.get((int) (fileIdx % pathList.size())) + File.separator + UtilAll.offset2FileName(createOffset);
```

Comment:
Here，Average modulus can be a default Realization，you can define a interface to improve scalability and then users also can implement the strategy by themself.It will be better!

Reply:
Please checkout the latest commit with MultiPathMappedFileQueue.java :)